api/repositories/droplet_repository.go

```diff
@@ -87,6 +87,40 @@ func (r *DropletRepo) GetDroplet(ctx context.Context, authInfo authorization.Inf
 	return cfBuildToDroplet(build)
 }
 
+func (r *DropletRepo) GetDroplets(ctx context.Context, authInfo authorization.Info, appGuid string) ([]DropletRecord, error) {
```

Comment:
`GetDroplets` repository method has the same essence as [`ListDroplets`](https://github.com/cloudfoundry/korifi/blob/d0483c8b89cc8112f731f882a8dfe1485f0b6cb5/api/repositories/droplet_repository.go#L156). Instead of implementing a new method, I would extend [`ListDropletsMessage`](https://github.com/cloudfoundry/korifi/blob/d0483c8b89cc8112f731f882a8dfe1485f0b6cb5/api/repositories/droplet_repository.go#L71) with `AppGUIDs []string` and make `ListDroplet` filter droplets by app guids if specified in on the message. 

The the [cfbuild webhook](https://github.com/cloudfoundry/korifi/blob/d0483c8b89cc8112f731f882a8dfe1485f0b6cb5/controllers/api/v1alpha1/cfbuild_webhook.go#L45) sets the `korifi.cloudfoundry.org/app-guid` label on the CFBuild to the app guid. Therefore, we could optimise the requests to leverage [set-based label selectors](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#set-based-requirement) in order not to filter collected droplets by app guids as an additional step.

Here is an [example](https://github.com/cloudfoundry/korifi/blob/d0483c8b89cc8112f731f882a8dfe1485f0b6cb5/controllers/controllers/workloads/processes/controller.go#L89) how to use label selectors when listing.